tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,543 @@
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+from numpy import equal
+import pytest
+
+from homeassistant.components import azure_data_explorer
+from homeassistant.components.azure_data_explorer.__init__ import AzureDataExplorer
+from homeassistant.components.azure_event_hub.const import CONF_SEND_INTERVAL, DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.core import State
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .conftest import FilterTest
+from .const import BASE_CONFIG_FULL, BASIC_OPTIONS, IMPORT_CONFIG
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def test_import(hass):
+    """Test the popping of the filter and further import of the config."""
+    config = {
+        DOMAIN: {
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
+    config[DOMAIN].update(IMPORT_CONFIG)
+    assert await async_setup_component(hass, DOMAIN, config)
+
+
+async def test_filter_only_config(hass):
+    """Test the popping of the filter and further import of the config."""
+    config = {
+        DOMAIN: {
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
+    assert await async_setup_component(hass, DOMAIN, config)
+
+
+async def test_unload_entry(hass, entry, mock_managed_streaming):
+    """Test being able to unload an entry.
+
+    Queue should be empty, so adding events to the batch should not be called,
+    this verifies that the unload, calls async_stop, which calls async_send and
+    shuts down the hub.
+    """
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    mock_managed_streaming.add.assert_not_called()
+    assert entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_failed_test_connection(hass):
+    """Test Error when no getting proper connection."""
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    assert entry.state == ConfigEntryState.LOADED
```

Comment:
Does this actually test a failed test_connection, if that happens in the init of the hub, then it shouldn't end up as loaded, right?